packages/edit-site/src/components/global-styles/use-global-styles-output.js

```diff
@@ -270,7 +292,11 @@ export function getStylesDeclarations(
 		const cssProperty = rule.key.startsWith( '--' )
 			? rule.key
 			: kebabCase( rule.key );
-		output.push( `${ cssProperty }: ${ compileStyleValue( rule.value ) }` );
+		let ruleValue = rule.value;
+		if ( typeof ruleValue !== 'string' ) {
```

Comment:
```suggestion
		if ( typeof ruleValue !== 'string' && ruleValue?.ref ) {
```

We could avoid assuming it's a ref here as well.